### ISSUE TITLE
DataFrame.hist / Series.hist argument inconsistency

### ISSUE THREAD

**Issue by wavexx** [pydata/pandas#11333]
Series.hist accepts arguments such as xlim, ylim because it uses Series.plot internally, but DataFrame.hist doesn't, as it forwards the arguments to pyplot.hist directly.

I don't know about other arguments, but I expected at least xlim/ylim to work.

**Comment by TomAugspurger** [comment]
Do you have an example, and what's your pandas version? These both work for me

```python
In [62]: s = pd.Series(np.random.randn(100))

In [63]: s.plot.hist(xlim=(0, 1), ylim=(-1, 1))
Out[63]: <matplotlib.axes._subplots.AxesSubplot at 0x113208dd8>

In [64]: s.to_frame().plot.hist(xlim=(0, 1), ylim=(-1, 1))
Out[64]: <matplotlib.axes._subplots.AxesSubplot at 0x113e25668>
```

Ahh, I see you're talking about the `hist` attached directly to the objects.

```python
In [83]: s.hist(xlim=(0, 1))
```

raises an AttributeError in matplotlib (1.5) for me.

**Comment by wavexx** [comment]
On 15/10/15 14:40, Tom Augspurger wrote:
> Ahh, I see you're talking about the `hist` attached directly to the objects.
> 
> ```python
> In [83]: s.hist(xlim=(0, 1))
> ```
> 
> raises an AttributeError in matplotlib (1.5) for me.

Precisely, as pyplot.hist doesn't have xlim/ylim (was it different for
1.4?).

It's not a deal breaker, but I'd rather uniform the various arguments
among plotting functions directly in pandas if it makes sense.

**Comment by TomAugspurger** [comment]
I don't think that `Series.hist` and `DataFrame.hist` have been deprecated, but we're pushing people towards using `.plot` where the arguments are consistent.

I'm still not able to reproduce your original problem. The oldest versions I have lying around are pandas 0.16.2 and matplotlib 1.4.3 `Series.hist(xlim=(0, 1))` raises and AttributeError, just like pandas 0.17 and matplotlib 1.5 do.

**Comment by Vimos** [comment]
With 
`pandas=0.18.1
matplotlib=1.5.1`

Code
`ax = df.gap.plot.hist(bins=100, xlim=[-20,20])`
works.

Code
`ax = df.gap.hist(bins=100, xlim=[-20,20])`
reports error
`AttributeError: Unknown property xlim`

This is confusing.
